executor/hash_table.go

```diff
@@ -73,17 +77,28 @@ func (hc *hashContext) initHash(rows int) {
 }
 
 // hashRowContainer handles the rows and the hash map of a table.
-// TODO: support spilling out to disk when memory is limited.
 type hashRowContainer struct {
-	records   *chunk.List
-	hashTable *rowHashMap
-
 	sc   *stmtctx.StatementContext
 	hCtx *hashContext
+
+	// hashTable stores the map of hashKey and RowPtr
+	hashTable *rowHashMap
+
+	// memTracker is the reference of records.GetMemTracker().
+	memTracker *memory.Tracker
```

Comment:
How about directly using `records.GetmemTracker()` when it's needed?

Reply:
`records` would be set to `nil` when `c.exceeded` is true for garbage collection.